libevm/SmartVM.cpp

```diff
@@ -93,18 +83,11 @@ namespace
 
 		~JitWorker()
 		{
-			DEV_GUARDED(x_mutex)
-				m_finished = true;
-			m_cv.notify_one();
+			push(JitTask::createStopSentinel());
 			m_worker.join();
```

Comment:
If we want more than std::queue provides we should drop the adapter. And I will do that.